include/wlr/types/wlr_xdg_shell.h

```diff
@@ -14,6 +14,13 @@ struct wlr_xdg_shell {
 	struct wl_listener display_destroy;
 
 	struct {
+		/**
+		 * The `new_surface` event signals that a client has requested to
+		 * create a new shell surface. At this point, the surface is ready to
+		 * be configured but not ready to be managed by the compositor by
+		 * adding it to a list of views. The surface will be ready to be
+		 * managed on the `map` event.
```

Comment:
More than that. I found that if you try to give the surface keyboard focus on the `new_surface` event instead of map, `weston-terminal` has not bound to a seat at that point so you can't send keyboard enter.

Reply:
Yeah, you're right. In this case, can you mention that the surface is unmapped when this event is emitted?